src/ast/semantic_analyser.cpp

```diff
@@ -1259,16 +1259,40 @@ bool SemanticAnalyser::is_final_pass() const
   return pass_ == num_passes_;
 }
 
-bool SemanticAnalyser::check_assignment(const Call &call, bool want_map, bool want_var)
+bool SemanticAnalyser::check_assignment(const Call &call, bool want_map, bool want_var, bool want_map_key)
 {
-  if (want_map && want_var)
+  if (want_map && want_var && want_map_key)
```

Comment:
Is is possible to want an assignment to a map, variable, _and_ map key?

Reply:
Nevermind, realizing now `check_assignment(call, true, true, true)` reads as "it's ok to assign me to a map/variable or use me as a map key".